backend/server/auth.js

```diff
@@ -0,0 +1,135 @@
+      const newUser = {
+        email: req.body.user_email,
+        hashed_password: encryption.hash,
+        salt: encryption.salt,
+        iterations: encryption.iterations,
+        dog_name: req.body.dog_name,
+        breed_id: result.id,
+        size: req.body.size,
+        dog_friendly: req.body.dog_friendly,
+        people_friendly: req.body.people_friendly,
+        energy: req.body.energy,
+        city: req.body.city,
+        state: req.body.state,
+        latitude: req.body.latitude,
+        longitude: req.body.longitude,
+        bio: req.body.bio,
+      };
+
+      User.create(newUser)
+        .then((user) => {
+          for (var i = 0; i < req.body.photos.length; i ++) {
+            let photo = {
+              user_id: user.id,
+              url: req.body.photos[i],
+            };
+            Photo.create(photo);
+          }
+          res.sendStatus(201);
+        })
+        .catch((err) => {
+          console.log(err);
+          res.sendStatus(400);
+        });
+    });
+  },
+
+  verify_email: function (req, res) {
+    // TODO: Change to findOne since email should be unique. (We should also maybe consider making email a unique clause in our db, but not a priority ¯\_(ツ)_/¯)
+    const { user_email } = req.query;
+    User.findOne({
+      attributes: ['email'],
+      where: {
+        email: user_email,
+      },
+    }).then((results) => {
+      if (!results) {
+        res.send('true');
+      } else {
+        res.send('false');
+      }
+    });
+  },
+
+  login: function (req, res) {
+    const { user_email, hashed_password_attempt } = req.query;
+    // TODO: Change to findOne since email should be unique.
+    User.findOne({
+      attributes: ['id', 'hashed_password', 'salt', 'iterations'],
+      where: {
+        email: user_email,
+      },
+    }).then((result) => {
+      if (!result) {
+        res.send('incorrect email');
+      } else {
+        const { hashed_password, salt, iterations } = result;
+        if (isPasswordCorrect(hashed_password, salt, iterations, hashed_password_attempt)) {
+          // TODO: Change to findOne since email should be unique
+          User.findOne({
+            where: {
+              email: user_email,
+            },
+            include: [
+              {
+                model: Breed,
+                attributes: {
+                 exclude: ['createdAt', 'updatedAt'],
+                },
+              },
+              {
+                model: Photo,
+                attributes: {
+                  exclude: ['createdAt', 'updatedAt'],
+                },
+              },
+            ],
+            attributes: {
+              exclude: ['hashed_password', 'salt', 'iterations', 'createdAt', 'updatedAt'],
+            },
+          }).then((userData) => res.send(userData));
+        } else if (!isPasswordCorrect(hashed_password, salt, iterations, hashed_password_attempt)) {
+          res.send('incorrect password');
+        }
```

Comment:
Please add error handling to this controller

Reply:
done!